ext/standard/var.c

```diff
@@ -992,6 +1015,24 @@ static void php_var_serialize_intern(smart_str *buf, zval *struc, php_serialize_
 				zend_bool incomplete_class;
 				uint32_t count;
 
+				if (ce->ce_flags & ZEND_ACC_ENUM) {
+					PHP_CLASS_ATTRIBUTES;
+
+					zval tmp;
+					zval *case_name_zval = zend_read_property_ex(ce, Z_OBJ_P(struc), ZSTR_KNOWN(ZEND_STR_CASE), 0, &tmp);
+
+					PHP_SET_CLASS_ATTRIBUTES(struc);
+					smart_str_appendl(buf, "E:", 2);
+					smart_str_append_unsigned(buf, ZSTR_LEN(class_name) + strlen(":") + Z_STRLEN_P(case_name_zval));
+					smart_str_appendl(buf, ":\"", 2);
+					smart_str_append(buf, class_name);
+					smart_str_appendl(buf, ":", 1);
```

Comment:
nit: `smart_str_appendc(buf, ':')` is common for single-char appending elsewhere and slightly more efficient in some builds